### PR TITLE
Spiff up virtual object UI experience

### PR DESCRIPTION
**Pull request by mjgiarlo**
## Why was this change made?

Improve the virtual object experience for users by:

1. Adding a facet to find all virtual objects; and
2. Rendering a pill on the show page to let users know they're looking at a virtual object

### Facet UI

Yes, there's only one virtual object in QA. Happy to take suggestions for how to label these better. @andrewjbtw. Both the collapsible heading *and* the link could both say `Virtual Objects` for instance? 🤷🏻‍♂️ 

Screenshot from 2022-02-14 16-25-27

### Show Page

Screenshot from 2022-02-14 16-24-51

## How was this change tested?

CI, QA

## Which documentation and/or configurations were updated?

None

### CODE REVIEW

File: app/assets/stylesheets/show.scss
Comment (mjgiarlo) [inline app/assets/stylesheets/show.scss:11]: Pill must be wider to render `VIRTUAL OBJECT`

File: app/assets/stylesheets/show.scss
Comment (mjgiarlo) [inline app/assets/stylesheets/show.scss:24]: Color chosen because it's in the brand guide and it's visually distinct from the other object type colors. Defer to @andrewjbtw and @astridu as to whether that's the best color to use here.

File: app/components/document_title_component.html.erb
Comment (mjgiarlo) [inline app/components/document_title_component.html.erb:8]: Now that we have an object type with a space in it (`virtual object`), we need a wee routine for deriving the class from the label.

File: app/controllers/catalog_controller.rb
Comment (mjgiarlo) [inline app/controllers/catalog_controller.rb:120]: I'm not sure if `Has Constituents` is the best label here. Defer to @andrewjbtw

File: app/controllers/catalog_controller.rb
Comment (mjgiarlo) [inline app/controllers/catalog_controller.rb:117]: I'm not sure if `Virtual Objects` is the best label here. Defer to @andrewjbtw

**Comment by astridu** [comment]
Can we make it a white box with black border? Thank you!

**Comment by mjgiarlo** [comment]
@astridu 💬 
> Can we make it a white box with black border? Thank you!

For sure. Thanks!

Currently, the other pills (for items, agreements, APOs, collections) do not have borders. Maybe we need a design issue around all five that we can apply at the same time in a later PR? Tagging @andrewjbtw to assess & prioritize.

File: app/assets/stylesheets/show.scss
Comment (andrewjbtw) [reply app/assets/stylesheets/show.scss:24]: I was not expecting to go all the way to a new page type with a pill. I thought we were just making a new facet.

Comment (mjgiarlo) [reply app/assets/stylesheets/show.scss:24]: This commit adds both the facet and the show page pill. It doesn't need to, though, if we're not ready or if this isn't the design we want.

Comment (andrewjbtw) [reply app/assets/stylesheets/show.scss:24]: I'm not opposed to going down this path, I just haven't thought through the implications. It does mean we would be able to take out the "download" links in the Contents section, which don't work for virtual objects since their content is metadata that links to content.

**Comment by jcoyne** [comment]
@mjgiarlo can you rebase this?

File: app/controllers/catalog_controller.rb
Comment (andrewjbtw) [reply app/controllers/catalog_controller.rb:120]: Maybe this is a theoretical question, but is it possible for something to tick the "I'm a virtual object" box and not have constituents? If so, having two values "has constituents" and "empty" could help with QA to identify empty objects. But if taking apart a virtual object removes it from the facet then I'd prefer the simplicity of "Virtual Object" - even if that ends up being redundant.

File: app/controllers/catalog_controller.rb
Comment (andrewjbtw) [reply app/controllers/catalog_controller.rb:117]: This is probably getting ahead of things but if we were to start indexing other special case objects, I could see using a more generic title for the facet. The other special case I can think of are digital serials, which are items that share a title and catkey but have their own druids, like issues of the same magazine.

In that scenario you could imagine a facet header with some name to be determined and then under it you'd see:

- Virtual Object
- Digital Serial

but that might just be making more work for ourselves at this stage.

File: app/controllers/catalog_controller.rb
Comment (mjgiarlo) [reply app/controllers/catalog_controller.rb:120]: I don't think that is possible here. This is a list of things in solr that have constituents. So there's a label for the facet header (currently `Virtual Objects`) and there's a label for the query (currently `Has Constituents`). Do you want one or both of these to change to `Virtual Object`?

File: app/controllers/catalog_controller.rb
Comment (mjgiarlo) [reply app/controllers/catalog_controller.rb:117]: These labels are super easy to change with very little in the way of consequences, so whatever works for both labels right now is fine by me!

File: app/controllers/catalog_controller.rb
Comment (andrewjbtw) [reply app/controllers/catalog_controller.rb:120]: Ok, let's just go with "Virtual Objects" in both and not overthink it. I can already see a huge benefit in having the facet.

Comment (mjgiarlo) [reply app/controllers/catalog_controller.rb]: @andrewjbtw OK, changed! Thank you.

**Comment by mjgiarlo** [comment]
@jcoyne @andrewjbtw OK, made the labeling change. We can consider this unheld whenever y'all want.

**Comment by jcoyne** [comment]
@mjgiarlo I've marked is as "HOLD" until Andrew has tested it on stage and he wants it to go to prod.

**Comment by mjgiarlo** [comment]
Unholding per conversation with @andrewjbtw

File: spec/models/solr_document_spec.rb
Comment (jcoyne) [inline spec/models/solr_document_spec.rb:29]: Is there a reason you removed this test?

**Review by jcoyne — CHANGES_REQUESTED** [review]

File: spec/models/solr_document_spec.rb
Comment (mjgiarlo) [reply spec/models/solr_document_spec.rb:29]: @jcoyne rebase mishap. I'll fix it in the morning. Good 👁️

Comment (mjgiarlo) [reply spec/models/solr_document_spec.rb]: Restored.

**Comment by mjgiarlo** [comment]
@jcoyne I fixed the rebase goof. Have another look-see?

**Comment by mjgiarlo** [comment]
I assume the build is failing because cocina updates are in progress: the DSA container has already updated and #3157 is pending. Not sure it's worth holding this PR up because of that. 🤷🏻‍♂️

**Review by jcoyne — APPROVED** [review]

**Pull request merged**
